nerdlets/e2m-gui-nerdlet/form-components/attribute-dropdown-selector.js

```diff
@@ -1,131 +1,124 @@
+        this.props.selectedAggregator !== prevProps.selectedAggregator ||
+        this.props.selectedEventType !== prevProps.selectedEventType) &&
+      !this.state.fetching
+    ) {
+      this.fetchData(this.props.selectedAccountID);
+    }
+  }
 
-	fetchData(selectedAccountID) {
-		if (
-			!selectedAccountID ||
-			!this.props.selectedAggregator ||
-			!this.props.selectedEventType
-		) {
-			return;
-		}
-		this.setState({ fetching: true });
-		NerdGraphQuery.query(
-			buildAttributeQueries(
-				selectedAccountID,
-				this.props.selectedEventType
-			)
-		)
-			.then(({ data, error }) => {
-				let attributeSet = new Set();
-				Object.keys(data["actor"])
-					.filter(i => i.includes("query"))
-					.forEach(query => {
-						try {
-							data["actor"][query]["nrql"]["results"].forEach(
-								attributeObj => attributeSet.add(attributeObj)
-							);
-						} catch (error) {
-							this.setState({
-								error: `${error}\n${JSON.stringify(data)}`
-							});
-						}
-					});
-				const attributes = Array.from(attributeSet).sort();
+  fetchData(selectedAccountID) {
+    if (
+      !selectedAccountID ||
+      !this.props.selectedAggregator ||
+      !this.props.selectedEventType
+    ) {
+      return;
+    }
+    this.setState({ fetching: true });
+    NerdGraphQuery.query(
+      buildAttributeQueries(selectedAccountID, this.props.selectedEventType)
+    )
+      .then(({ data }) => {
+        const attributeSet = new Set();
+        Object.keys(data.actor)
+          .filter(i => i.includes('query'))
+          .forEach(query => {
+            try {
+              data.actor[query].nrql.results.forEach(attributeObj =>
+                attributeSet.add(attributeObj)
+              );
+            } catch (error) {
+              this.setState({
+                error: `${error}\n${JSON.stringify(data)}`
+              });
+            }
+          });
+        const attributes = Array.from(attributeSet).sort();
 
-				this.setState({
-					attributes,
-					fetching: false,
-					error: null
-				});
-			})
-			.catch(error => this.setState({ error, fetching: false }));
-	}
+        this.setState({
+          attributes,
+          fetching: false,
+          error: null
+        });
+      })
+      .catch(error => this.setState({ error, fetching: false }));
+  }
 
-	render() {
-		// Since multiple accounts are selected, the event types shown below are aggregated from
-		//      all accounts. Each individual account may not have the event types here.
-		if (this.state.error) {
-			console.log(this.state.error);
-			return <div>Check console for error</div>;
-		} else if (this.state.fetching) {
-			return <Spinner />;
-		}
+  render() {
+    // Since multiple accounts are selected, the event types shown below are aggregated from
+    //      all accounts. Each individual account may not have the event types here.
+    if (this.state.error) {
+      console.log(this.state.error); // eslint-disable-line no-console
```

Comment:
Rationale?

Reply:
I made the assumption in the linting rules that any log of an error was an acceptable exception and added the appropriate rules.